crates/wasmtime/src/module.rs

```diff
@@ -241,6 +241,10 @@ impl Module {
                     if #[cfg(feature = "wat")] {
                         let mut e = e.downcast::<wat::Error>()?;
                         e.set_path(file);
+                        if e.to_string().ends_with("input bytes aren't valid utf-8") {
+                            eprintln!("note: wasmtime might be trying to load a precompiled binary without --allow-compiled.");
```

Comment:
I don't think the wasmtime library should print to stderr. There is no way for a binary to suppress this.

Reply:
You're right, wasmtime is a library so eprintln should be avoided if possible. How about deferring this eprintln to the wasmtime-cli? This would make wasmtime-cli add `wat` dependency to downcast error.
Also the `#[cfg(feature = "wat")]` branch condition cannot be used on wasmtime-cli, so this eprintln would be fired unconditionally.

Reply:
I've just suggested a different way to report this error, so let's delete this `eprintln` in favor of that alternative.